src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -365,85 +311,65 @@ namespace ts.codefix {
-        const originalNodeParent = node.original ? node.original.parent : node.parent;
-        if (prevArgName && !shouldReturn && (!originalNodeParent || isPropertyAccessExpression(originalNodeParent))) {
-            return createTransformedStatement(prevArgName, createAwait(node), transformer);
-        }
-        else if (!prevArgName && !shouldReturn && (!originalNodeParent || isPropertyAccessExpression(originalNodeParent))) {
-            return [createStatement(createAwait(node))];
+    /**
+     * Transforms the 'x' part of `x.then(...)`, or the 'y()' part of `y.catch(...)`, where 'x' and 'y()' are Promises.
+     */
+    function transformPromiseExpressionOfPropertyAccess(node: Expression, transformer: Transformer, prevArgName?: SynthBindingName): readonly Statement[] {
+        if (shouldReturn(node, transformer)) {
+            return [createReturn(getSynthesizedDeepClone(node))];
         }
 
-        return [createReturn(getSynthesizedDeepClone(node))];
+        return createVariableOrAssignmentOrExpressionStatement(prevArgName, createAwait(node));
     }
 
-    function createTransformedStatement(prevArgName: SynthBindingName | undefined, rightHandSide: Expression, transformer: Transformer): readonly Statement[] {
-        if (!prevArgName || isEmpty(prevArgName)) {
+    function createVariableOrAssignmentOrExpressionStatement(variableName: SynthBindingName | undefined, rightHandSide: Expression): readonly Statement[] {
+        if (!variableName || isEmptyBindingName(variableName)) {
             // if there's no argName to assign to, there still might be side effects
-            return [createStatement(rightHandSide)];
+            return [createExpressionStatement(rightHandSide)];
         }
 
-        if (isSynthIdentifier(prevArgName) && prevArgName.types.length < prevArgName.numberOfAssignmentsOriginal) {
+        if (isSynthIdentifier(variableName) && variableName.hasBeenDeclared) {
             // if the variable has already been declared, we don't need "let" or "const"
-            return [createStatement(createAssignment(getSynthesizedDeepClone(prevArgName.identifier), rightHandSide))];
+            return [createExpressionStatement(createAssignment(getSynthesizedDeepClone(variableName.identifier), rightHandSide))];
         }
 
-        return [createVariableStatement(/*modifiers*/ undefined,
-            (createVariableDeclarationList([createVariableDeclaration(getSynthesizedDeepClone(getNode(prevArgName)), /*type*/ undefined, rightHandSide)], getFlagOfBindingName(prevArgName, transformer.constIdentifiers))))];
+        return [
+            createVariableStatement(
+                /*modifiers*/ undefined,
+                createVariableDeclarationList([
+                    createVariableDeclaration(
+                        getSynthesizedDeepClone(getNode(variableName)),
+                        /*type*/ undefined,
+                        rightHandSide)],
+                    NodeFlags.Const))];

```

Comment:
I guess I'm a little surprised this always comes out `const`, but if nothing breaks...